addons/pos_loyalty/static/src/js/Loyalty.js

```diff
@@ -471,10 +471,12 @@ patch(Order.prototype, "pos_loyalty.Order", {
         const rewardLines = [];
         const nonRewardLines = [];
         for (const line of orderlines) {
-            if (line.is_reward_line) {
-                rewardLines.push(line);
-            } else {
-                nonRewardLines.push(line);
+            if (line) {
+                if (line.is_reward_line) {
+                    rewardLines.push(line);
+                } else {
+                    nonRewardLines.push(line);
+                }
```

Comment:
Linked to your PR? Can you explain the change in an other commit ? Seems to be a bug ?

Reply:
Yes it's a bug in orderlines when one is removed, the array is not fully updated. I added the explanation in the commit (not in another because it's difficult to come back to a previous commit to edit it when you have more than 1 commit).